src/lib/core.d.ts

```diff
@@ -836,7 +836,7 @@ interface RegExp {
     lastIndex: number;
 
     // Non-standard extensions
-    compile(): RegExp;
+    compile(): this;

```

Comment:
I don't think there is a guarantee that compile returns your subclass if you subclass RegExp, so it should return plain `RegExp` instead.